### PR TITLE
ASoC: Intel: soc-acpi: use compatible IDs

### PR DESCRIPTION
**Pull request by plbossart**
Minor improvements on patches suggested by @brentlu upstream (use of const, ~~ARRAY_SIZE~~) and support for BYT/CHT.

### CODE REVIEW

File: sound/soc/intel/common/soc-acpi-intel-adl-match.c [hunk 1]
```diff
@@ -285,25 +285,30 @@ static const struct snd_soc_acpi_codecs adl_max98360a_amp = {
 	.codecs = {"MX98360A"}
 };
 
+static const struct snd_soc_acpi_codecs adl_rt5682_rt5682s_hp = {
+	.codecs = {"10EC5682", "RTL5682"},
+	.num_codecs = ARRAY_SIZE(adl_rt5682_rt5682s_hp.codecs),
```
Comment (brentlu) [inline sound/soc/intel/common/soc-acpi-intel-adl-match.c]: It's always 3 on my device.

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-adl-match.c]: Not sure what you are saying @brentlu, which is it
a) you have 3 codecs and there's one missing in the list
b) ARRAY_SIZE does not give the expected result of 2?

Comment (brentlu) [reply sound/soc/intel/common/soc-acpi-intel-adl-match.c]: the b. the result is always 3 since the .codecs is a fixed 2-d array.

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-adl-match.c]: Ah yes, thanks @brentlu. I thought this was a variable array initialized, but nope:

````c
#define SND_SOC_ACPI_MAX_CODECS 3

struct snd_soc_acpi_codecs {
	int num_codecs;
	u8 codecs[SND_SOC_ACPI_MAX_CODECS][ACPI_ID_LEN];
};
````

I guess there's no other choice but to add the num_codecs manually. will update.

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-adl-match.c]: updated

File: sound/soc/intel/common/soc-acpi-intel-cht-match.c [hunk 2]
```diff
@@ -118,7 +115,7 @@ struct snd_soc_acpi_mach  snd_soc_acpi_intel_cherrytrail_machines[] = {
 		.sof_tplg_filename = "sof-cht-da7213.tplg",
```
Comment (brentlu) [inline sound/soc/intel/common/soc-acpi-intel-cht-match.c]: The entry "DLGS7212" is redundant.

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-cht-match.c]: indeed, thanks for spotting this. fixed now.

**Comment by plbossart** [comment]
@brentlu if you want to the patches directly upstream that's fine with me. I am running behind on other things. Thanks!

**Comment by brentlu** [comment]
OK. I will send a V3 patch series to alsa-dev including all 6 patches.

**Pull request merged**